File: 1017-convert-to-base-2/1017-convert-to-base-2.cpp
#include <string>
#include <algorithm>
class Solution {
public:
    string baseNeg2(int n) {
        if (n==0) return "0";
        int base = -2;
        string ans;
        while (n!=0){
            int rem = n % base;
            n = n / base;
            if (rem < 0){
                
                rem = rem + 2;
                n += 1;
            }
            
            ans += to_string(rem);
     
        }
            cout<<ans;
            reverse(ans.begin(),ans.end());
            return ans;

    }

};

File: 1017-convert-to-base-2/1017-convert-to-base-2.py
class Solution:
    def baseNeg2(self, n: int) -> str:
        if n == 0 : return "0"
        base = -2
        ans = ""
        while(n!=0):
            rem = n % base 
            n = n // base  #dont use n = int(n/base)  5 // -2 = -3   (integer floor division) 5 / -2 = -2.5 int(-2.5) = -2   "// in Python rounds down toward −∞ but int() truncates toward 0"
            if (rem < 0):
                rem += 2
                n += 1
            ans += str(rem)
        ans = ans[::-1]
        return ans
